GeometryCore/src/Polygon.cpp

```diff
@@ -130,24 +130,30 @@ namespace geom_utils
     Polygon Polygon::convexHull() const 
     { 
         if (points.size() <= 3) return *this; //can't bild a polygon from 2 points or less
+
+        auto my_cross = [](const FPoint2D& O, const FPoint2D& A, const FPoint2D& B) {return (A.x - O.x) * (B.y - O.y) - (A.y - O.y) * (B.x - O.x); };
```

Comment:
Please dont re-write things twice. We have already implemented vector cross product function - please see LinearAlgebra.